sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -370,6 +372,17 @@ private Task ExecuteOperationAsync(MobileServiceTableOperation operation, JObjec
 
                 try
                 {
+                    if (MobileServiceTableOperationKind.Insert != operation.Kind)
+                    {
+                        // Refresh the item's version, in case it changed while we were awaiting the item lock
+                        var localStoreItem = await this.Store.LookupAsync(operation.TableName, operation.ItemId);
```

Comment:
Did you get a chance to look at the document I linked to in issue #615? It's much more descriptive and complete in terms of the various problems that occur due to the race condition.

Regarding the versioning issue, now that I re-read your initial comment, are you suggesting that my fix could cause us to stomp on server changes in the single-client case? I realize it's bad in the case where the first push causes an actual conflict, where the server really has a future version from another client updating it, but I'm not sure I see how we could trample changes to other columns made by the operation(s) in the first push, because the change only affects what is written locally when the operation gets a chance to execute and first wants to effect the change to the local store.  Later, the item is pulled from the store in the PushAction by calling LoadOperationItem(), and there it will have all the columns as they were when we returned from the last remote execution in the previous push, but with the last __version we know about from the server.  If no other client changes that row, this seems like the correct behavior.  Only once the row in the local store has been tainted with server data (from a pull or a merge conflict) is this incorrect, and in that case all pending operations have to be invalidating.

I'm investigating a workaround for now for the later case: I plan to keep all my conflict resolution code, which is pretty complicated already without these issues that cropped up, but in the case where I resolve a conflict by picking the server item, I was hoping to maybe use the CancelAndDiscard*() API, except that is only available in OnPushComplete(), so somehow I'd have to know that in the SyncHandler I need to fail all subsequent operations *for this push*, but I don't think we have the context in SyncHandler to know if we're in one push or another...plus due to the race this change fixes, you can keep adding operations to the OperationQueue even while a push is in-flight.

I'm curious: is this stuff being actively discussed internally? Are these known issues you're working on, or is this back-burner stuff? I'm wondering how much effort to put into working around these issues in client code, and/or how much time to devote to fixing it in my fork of the SDK, or if something is in the pipeline to be fixed soonish.  Thanks!